app/src/main/java/com/github/polybooks/FilteringBooksActivity.kt

```diff
@@ -0,0 +1,145 @@
+package com.github.polybooks
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import android.widget.CheckBox
+import android.widget.RadioButton
+import android.widget.RadioGroup
+import androidx.appcompat.app.AppCompatActivity
+import com.github.polybooks.core.Course
+import com.github.polybooks.core.Field
+import com.github.polybooks.core.Interest
+import com.github.polybooks.core.Semester
+import com.github.polybooks.core.database.implementation.DummyBookQuery
+import com.github.polybooks.core.database.implementation.DummySalesQuery
+import com.github.polybooks.core.database.interfaces.BookOrdering
+import com.github.polybooks.core.database.interfaces.BookQuery
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * This activity let the users to select the sorting and filtering parameters
+ * by clicking on the corresponding value (eg. "Architecture" button for
+ * parameter "Field"). If no value is selected for a given parameter, then the
+ * query won't be filtered on this parameter
+ *
+ * Note : should implement a dynamic version of it soon
+ */
+class FilteringBooksActivity : AppCompatActivity() {
+
+    private lateinit var mResetParameters : Button
+    private lateinit var mResults : Button
+
+    //--- TODO hardcoded parameters: make it dynamic
+    private lateinit var mSortGroup : RadioGroup
+    private lateinit var mSortPopularity : RadioButton
+    private lateinit var mSortTitleInc : RadioButton
+    private lateinit var mSortTitleDec : RadioButton
+
+    private lateinit var mFieldCS : CheckBox
+    private lateinit var mFieldBio : CheckBox
+    private lateinit var mFieldArchi : CheckBox
+
+    private lateinit var mSemBa1 : CheckBox
+    private lateinit var mSemBa2 : CheckBox
+    private lateinit var mSemBa3 : CheckBox
+    private lateinit var mSemMa1 : CheckBox
+    private lateinit var mSemMa2 : CheckBox
+
+    private lateinit var mCourseCS306 : CheckBox
+    private lateinit var mCourseCOM480 : CheckBox
+
+    //------
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_filtering_books)
+
+        // Get a reference to the UI parameters
+        mResetParameters = findViewById(R.id.reset_button)
+        mResults = findViewById(R.id.results_button)
+
+        // Set behaviour Reset and Results
+        setResetButtonBehaviour()
+        setResultsButtonBehaviour()
+
+        // --- TODO hardcoded : make it dynamic
+        setParametersButtons()
+    }
+
+    private fun setResetButtonBehaviour() {
+        mResetParameters.setOnClickListener {
```

Comment:
An alternative to setOnClickListener is to use the android:onClick in the xml.
I find it slightly cleaner, but ultimately both do exactly the same thing, so it's fine like that too.

Reply:
done :)